Assets/MRTK/Core/Utilities/Editor/PackageManifest/PackageManifestUpdater.cs

```diff
@@ -300,12 +298,148 @@ internal static void EnsureMSBuildForUnity()
                 manifestFileLines[packageLine] = $"    \"{MSBuildPackageName}\": \"{MSBuildPackageVersion}\",";
             }
 
-            // Update the manifest file.
+            // Update the manifest
+            if (!needToAddRegistry && !needToAddPackage)
+            {
+                // No changes required.
+                return;
+            }
+            WriteManifest(
+                manifest, 
+                manifestFileLines, 
+                registriesStartIndex,
+                registriesEndIndex);
+        }
+
+        /// <summary>
+        /// Removes the package manager settings that allow for using MSBuild for Unity.
+        /// </summary>
+        [MenuItem("Mixed Reality Toolkit/MSBuild/Remove MSBuild for Unity", priority = 100)]
+        internal static void DisableMSBuildForUnity()
+        {
+            List<string> manifestFileLines;
+            PackageManifest manifest = LoadManifest(out manifestFileLines);
+
+            if (manifest == null)
+            {
+                Debug.LogError($"Failed to read the package manifest file ({ManifestFilePath})");
+                return;
+            }
+
+            // Ensure that pre-existing scoped registries are retained.
+            List<ScopedRegistry> scopedRegistries = new List<ScopedRegistry>();
+            if ((manifest.scopedRegistries != null) && (manifest.scopedRegistries.Length > 0))
+            {
+                scopedRegistries.AddRange(manifest.scopedRegistries);
+            }
+
+            // Attempt to find an entry in the scoped registries collection for the MSBuild for Unity URL
+            int registryToRemove = -1;
+            for (int i = 0; i < scopedRegistries.Count; i++)
+            {
+                if (scopedRegistries[i].url == MSBuildRegistryUrl)
+                {
+                    registryToRemove = i;
+                    break;
+                }
+            }
+            if (registryToRemove != -1)
+            {
+                scopedRegistries.RemoveAt(registryToRemove);
+                // Update the manifest's scoped registries.
+                manifest.scopedRegistries = scopedRegistries.ToArray();
+            }
+
+            // Attempt to find the msbuild for unity package
+            // This loop also identifies the lines that enclose the scoped registry collection.
+            int registriesStartIndex = -1;
+            int registriesEndIndex = -1;
+            int packageToRemove = -1;
+            for (int i = 0; i < manifestFileLines.Count; i++)
+            {
+                if (manifestFileLines[i].Contains("\"scopedRegistries\":"))
+                {
+                    registriesStartIndex = i;
+                }
+                if (manifestFileLines[i].Contains("],") && (registriesStartIndex != -1) && (registriesEndIndex == -1))
+                {
+                    registriesEndIndex = i;
```

Comment:
What happens if we don't find the end (i.e. we don't find ], or there is ] , instead which is also perfectly valid json)?

Reply:
this was based on the output of Unity's json API.

Reply:
have a better (more robust) implementation now